source/simulator/stokes_matrix_free.cc

```diff
@@ -2928,139 +2904,14 @@ namespace aspect
-                    {
-                      fe_values_projection.reinit(DG_cell);
-                      fe_values_projection.get_function_values(level_viscosity_vector[level],
-                                                               dg_dof_indices,
-                                                               visc_on_quad);
-                    }
-
-                  for (unsigned int q=0; q<n_q_points; ++q)
-                    {
-                      const double viscosity = (dof_handler_projection.get_fe().degree == 0
-                                                ?
-                                                level_viscosity_vector[level](dg_dof_indices[0])
-                                                :
-                                                std::min(std::max(visc_on_quad[q], static_cast<GMGNumberType>(minimum_viscosity)),
-                                                         static_cast<GMGNumberType>(maximum_viscosity)));
-
-                      for (unsigned int k=0; k<dofs_per_cell; ++k)
-                        {
-                          symgrad_phi_u[k] = fe_values[velocities].symmetric_gradient (k, q);
-
-                          if (is_compressible)
-                            div_phi_u[k] = fe_values[velocities].divergence (k, q);
-                        }
-
-                      const double JxW = fe_values.JxW(q);
-                      for (unsigned int i=0; i<dofs_per_cell; ++i)
-                        for (unsigned int j=0; j<dofs_per_cell; ++j)
-                          {
-                            cell_matrix(i,j) += 2. * viscosity * (symgrad_phi_u[i]*symgrad_phi_u[j])
-                                                * JxW;
-
-                            if (is_compressible)
-                              cell_matrix(i,j) +=  (-2./3.) * viscosity * (div_phi_u[i]*div_phi_u[j])
-                                                   * JxW;
-                          }
-                    }
-
-                  cell->get_mg_dof_indices (local_dof_indices);
-
-                  boundary_constraints.distribute_local_to_global (cell_matrix,
-                                                                   local_dof_indices,
-                                                                   diagonal_matrix);
-                }
-
-            diagonal_matrix.compress(VectorOperation::add);
-
-            dealii::LinearAlgebra::distributed::Vector<GMGNumberType> diagonal;
-            // This assignment converts from type double to GMGNumberType.
-            diagonal = diagonal_matrix.get_vector();
-            mg_matrices_A_block[level].set_diagonal(diagonal);
-          }
-        else
-          {
-            mg_matrices_A_block[level].compute_diagonal();
-          }
+        mg_matrices_A_block[level].compute_diagonal();
 
         // This vector is no longer needed. Resize to 0.
-        level_viscosity_vector[level].reinit(0);
+        //level_viscosity_vector[level].reinit(0);
```

Comment:
I guess we can remove these two lines, and move `level_viscosity_vector[level].reinit(0);`  to `evaluate_material_model()` after `mg_matrices_A_block[level].set_cell_data (level_cell_data[level]);`